"""
  This program runs as a client that sends a series of messages 
  to a server using the reliable_sendmess function from the reliable library.
  
  It prints an affirmative message if everything went correctly.  
  Otherwise it prints the error that occurred.
  
  This program must be started after the server program is started.


  Arguments needed:
	serverip - the dot-decimal notation ip address of the server
	port - the port number of the server computer

"""



include reliable.repy


if callfunc == 'initialize':

  if len(callargs) != 2 :
    print 'Wrong number of call args used.  The test could not be run.'
    return

  # get the server ip address and port number
  serverip = callargs[0]
  port = int(callargs[1])

  # make message contents to send to the server 
  packet1 = "first packet"
  packet2 = "second packet"
  packet3 = "third packet"
  packet4 = "fourth packet"
 
  try:

    # send the messages using the reliable_sendmess from the library
    reliable_sendmess(serverip, port, packet1)
    reliable_sendmess(serverip, port, packet2)
    reliable_sendmess(serverip, port, packet3)
    reliable_sendmess(serverip, port, packet4)
    print "test completed"

  except Exception, e:
    print "test indeterminant:  " + str(e)

  exitall()
